vmdb/app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -40,7 +44,7 @@ def save_inventory(type, parent, hash)
     found = parent.send(type)
     if found.nil?
       found = parent.send("build_#{type}", hash.except(:id))
-      parent.send("#{type}=", found)
+      parent.update_attributes!(type => found)
```

Comment:
@Fryguy I think that in the first example running `s.computer_system.build_operating_system` is not setting `s.computer_system` as dirty so `save!` won't do anything.

In the second case you create everything before the final `save!` and it happens to work simply because the objects are dirty for other reasons (just created).